mesos-client/src/main/scala/com/mesosphere/mesos/client/MesosClient.scala

```diff
@@ -294,6 +300,11 @@ object MesosClient extends StrictLogging with StrictLoggingFlow {
       system: ActorSystem,
       materializer: ActorMaterializer): Source[MesosClient, NotUsed] = {
 
+    if (!isMultiRoleFramework(frameworkInfo)) {
```

Comment:
I would not put this check into the client. It's the "limitation" of our core-scheduler.

Reply:
okay, makes sense!